include/common/mir/input/keymap.h

```diff
@@ -20,58 +20,38 @@
 #ifndef MIR_INPUT_KEYMAP_H_
 #define MIR_INPUT_KEYMAP_H_
 
+#include <memory>
 #include <string>
-#include <ostream>
+
+struct xkb_keymap;
+struct xkb_context;
 
 namespace mir
 {
 namespace input
 {
+using XKBKeymapPtr = std::unique_ptr<xkb_keymap, void(*)(xkb_keymap*)>;
 
-struct Keymap
+class Keymap
 {
+public:
     Keymap() = default;
-    Keymap(std::string&& model,
-           std::string&& layout,
-           std::string&& variant,
-           std::string&& options)
-        : model{model}, layout{layout}, variant{variant}, options{options}
-    {
-    }
-
-    Keymap(std::string const& model,
-           std::string const& layout,
-           std::string const& variant,
-           std::string const& options)
-        : model{model}, layout{layout}, variant{variant}, options{options}
-    {
-    }
+    virtual ~Keymap() = default;
 
-    std::string model{"pc105+inet"};
-    std::string layout{"us"};
-    std::string variant;
-    std::string options;
+    virtual auto operator==(Keymap const& other) const -> bool = 0;
```

Comment:
On principle, I don't like `operator==()` on an interface. The meaning is never clear as **interfaces do not have value semantics**. What does it mean semantically for it to be `virtual`?

I would prefer a good function name.

Reply:
Done